### PR TITLE
adds episode selection to downloads

### PR DESCRIPTION
**Pull request by sandikbarr**
#24

### CODE REVIEW

File: src/app/shared/filter/canned-range.component.spec.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ describe('CannedRangeComponent', () => {
     expect(comp.filter.endDate.valueOf()).toEqual(comp.endOfTodayUTC().valueOf());
   });
 
-  it('should go to prev range when PREV button is click', () => {
+  xit('should go to prev range when PREV button is click', () => {
```
Comment (sandikbarr) [inline src/app/shared/filter/canned-range.component.spec.ts]: would be a good week to address this one: https://github.com/PRX/metrics.prx.org/issues/37

File: src/app/shared/filter/canned-range.component.ts [hunk 2]
```diff
@@ -101,7 +101,7 @@ export class CannedRangeComponent implements OnInit, OnDestroy {
     this.whenOptions.push([LAST_MONTH, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
 
     // first of this month - 1 day and 2 months
-    utcEndDate = this.endOfTodayUTC().date(1).subtract(1, 'days').subtract(2, 'months');
+    utcEndDate = this.endOfTodayUTC().date(1).subtract(2, 'months').subtract(1, 'days');
```
Comment (sandikbarr) [inline src/app/shared/filter/canned-range.component.ts]: Prior 3 months was ending on 7/30. Oops.

File: src/app/shared/util/metrics.util.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { PodcastModel, EpisodeModel, FilterModel, MetricsType, PodcastMetricsModel, EpisodeMetricsModel } from '../../ngrx/model';
+
```
Comment (sandikbarr) [inline src/app/shared/util/metrics.util.ts]: I separated these filtering and utility functions from the selectors for better organization.

File: src/app/shared/filter/episodes.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { selectEpisodeFilter, selectEpisodes } from '../../ngrx/reducers';
+import { EpisodeModel } from '../../ngrx/model';
+import { CastleFilterAction } from '../../ngrx/actions';
+
+@Component({
+  selector: 'metrics-episodes',
+  template: `
+    <span>Episodes:</span>
+    <span>
+      <prx-select [options]="allEpisodeOptions" [selected]="selected" searchable="true" (onSelect)="onEpisodesChange($event)"></prx-select>
```
Comment (cavis) [inline src/app/shared/filter/episodes.component.ts]: Seems like there are some UX issues with how the select works here.  May need to address them in styleguide at some point.

No results rendered until you start typing, so it's hard to see/alter the selections.  In this case, would probably be best if it rendered the "most recent N", but typing in the filter can get you to the older episodes.  Not sure what to do about the "selected but not visible per the filter" problem.

**Comment by sandikbarr** [comment]
@cavis I'm not clear on what you're asking me to change. I can add those canned options (not sure they were really shown to be helpful, given the last episode overwhelming the chart and producers only really looking at the most recent episode), but they're not going to solve the problem of options not being visible in `prx-select`.

**Comment by cavis** [comment]
No changes!  I just tend to use "request changes" to explicitly block merging while i ask questions.

Hashed out in slack - will merge this functionality, and open some other tickets to improve the upstream prx-select.

**Pull request merged**